client/src/main/java/io/pravega/client/stream/impl/CheckpointState.java

```diff
@@ -246,6 +255,7 @@ private void read00(RevisionDataInput input, CheckpointStateBuilder builder) thr
             builder.uncheckpointedHosts(input.readMap(stringDeserializer, in -> in.readCollection(stringDeserializer, ArrayList::new)));
             builder.checkpointPositions(input.readMap(stringDeserializer, in -> in.readMap(segmentDeserializer, longDeserializer)));
             builder.lastCheckpointPosition(input.readMap(segmentDeserializer, longDeserializer));
+            builder.lastStreamCutPosition(input.readMap(segmentDeserializer, longDeserializer));
```

Comment:
we should be backward compatible here, this line would throw exceptions when trying to read from data from an older client.

Reply:
I've added a new revision and made changes to `read00` ensure backward compatibility. I've also added a test for the same in `SerializationTest`.

Reply:
The field has been removed so there shouldn't be an issue in regards to backward compatibility.